GRPC fails to compile on Ubuntu

Generating cache.mk
[C]       Compiling src/core/client_config/subchannel.c
src/core/client_config/subchannel.c: In function ‘disconnect’:
src/core/client_config/subchannel.c:272:5: error: conversion of unsigned constant value to negative integer [-Werror=sign-conversion]
cc1: all warnings being treated as errors
make: *** [/home/developer/grpc-master/objs/opt/src/core/client_config/subchannel.o] Error 1

--- a/src/core/ext/client_config/subchannel.c
+++ b/src/core/ext/client_config/subchannel.c
@@ -268,7 +268,7 @@ static void disconnect(grpc_exec_ctx *exec_ctx, grpc_subchannel *c) {
   con = GET_CONNECTED_SUBCHANNEL(c, no_barrier);
   if (con != NULL) {
     GRPC_CONNECTED_SUBCHANNEL_UNREF(exec_ctx, con, "connection");
-    gpr_atm_no_barrier_store(&c->connected_subchannel, 0xdeadbeef);
+    gpr_atm_no_barrier_store(&c->connected_subchannel, (gpr_atm)0xdeadbeef);
   }
   gpr_mu_unlock(&c->mu);
 }

